tests/protocols/test_protocols.py

```diff
@@ -0,0 +1,50 @@
+import tempfile
+
+import mypy.api
+import pytest
+from ophyd import sim
+
+from ophyd_async import protocols as bs_protocols
+
+
+def test_readable():
+    assert isinstance(sim.motor1, bs_protocols.AsyncReadable)
+    assert isinstance(sim.det1, bs_protocols.AsyncReadable)
+    assert not isinstance(sim.flyer1, bs_protocols.AsyncReadable)
+
+
+def test_pausable():
+    assert isinstance(sim.det1, bs_protocols.AsyncPausable)
+
+
+# I think the commented out tests pass because __getattr__ is implemented, but not sure
+@pytest.mark.skip(reason="ophyd missing py.typed to communicate type hints to mypy")
+@pytest.mark.parametrize(
+    "type_, hardware, pass_",
+    [
+        ("Readable", "ABDetector(name='hi')", True),
+        ("Readable", "SynAxis(name='motor1')", True),
+        ("Readable", "TrivialFlyer()", False),
+        ("Configurable", "ABDetector(name='hi')", True),
+        ("Pausable", "ABDetector(name='hi')", True),
+    ],
+)
+def test_mypy(type_, hardware, pass_):
```

Comment:
This is already covered by the separate CI job to run mypy, so a test for it isn't needed